### PR TITLE
Laboratory work #1, Ivan Shmakov - 23FPL1

### PR DESCRIPTION
**Pull request by CloudWhisp**
Здравствуйте, Василиса Михайловна! Присылаю свои изменения в лабораторной работе! Заранее спасибо

### CODE REVIEW

**Comment by Vasilisa-Blyudova** [comment]
Good job! Don't forget to change the score in settings.json!

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -20,7 +19,13 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
-
+    token_list = []
+    if not isinstance(text, str):
+        return None
+    for letter in text.lower():
+        if letter.isalpha() and letter != 'º':
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:26]: use continue operator

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -64,6 +84,15 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not isinstance(predicted, list) or not isinstance(actual, list):
+        return None
+    if len(predicted) != len(actual):
+        return None
+    mse = 0
+    n = len(predicted)
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:92]: rename

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -105,6 +159,27 @@ def detect_language(
...
+        languages = [str(profile_1['name']), str(profile_2['name'])]
+        if not isinstance(languages, list):
+            return None
+        if languages is not None:
+            func_result = sorted(languages)[0]
+    elif first_metric < second_metric:
+        func_result = str(profile_1['name'])
+    else:
+        func_result = str(profile_2['name'])
+    if isinstance(func_result, str):
```
Comment (Vasilisa-Blyudova) [inline lab_1_classify_by_unigrams/main.py:180]: return None here

**Review by Vasilisa-Blyudova — COMMENTED** [review]
Very good job! Increase the rating to 10!

**Comment by Vasilisa-Blyudova** [comment]
Don't forget to reply to comments (fixed/done)

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py:92]: Done

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -5,8 +5,15 @@
 """
 # pylint:disable=too-many-locals, unused-argument, unused-variable
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py:7]: Done

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -5,8 +5,15 @@
...
 
+def tokenize(text):
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -5,8 +5,15 @@
...
+    token_list = []
+    if isinstance(text, str):
+        for i in text.lower():
+            if i.isalpha() and i != 'º':
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -34,6 +41,12 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    freq_dict = {}
+    if isinstance(tokens, list):
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -64,6 +82,13 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if isinstance(predicted, list) and isinstance(actual, list):
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -64,6 +82,13 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
+        mse = 0
+        n = len(predicted)
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -84,7 +109,23 @@ def compare_profiles(
     In case of corrupt input arguments or lack of keys 'name' and
     'freq' in arguments, None is returned
     """
-
+    if isinstance(unknown_profile, dict) and isinstance(profile_to_compare, dict):
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/start.py [hunk 11]
```diff
@@ -14,8 +15,10 @@ def main() -> None:
         de_text = file_to_read_de.read()
     with open("assets/texts/unknown.txt", "r", encoding="utf-8") as file_to_read_unk:
         unknown_text = file_to_read_unk.read()
-    result = None
+    language = 'en'
+    result = create_language_profile(language, en_text)
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/start.py]: Done

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -84,7 +109,23 @@ def compare_profiles(
...
+      if 'frec' in profile_to_compare and 'frec' in unknown_profile:
```
[the remark this answers is not in the document]
Comment (CloudWhisp) [reply lab_1_classify_by_unigrams/main.py]: Fixed

**Pull request closed without merging**